### ISSUE TITLE
Unable to build on mac m1

### ISSUE THREAD

**Issue by jhay06** [kivy/kivy-ios#613]
Hi I'm trying to create a kivy app for ios using mcbook air with m1 cpu 
but during toolchain build kivy , it fails with the output below

  STDOUT:
checking build system type... arm-apple-darwin20.4.0
checking host system type... arm-apple-darwin20.4.0
checking for python3.8... no
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "darwin"
checking for gcc... clang -Qunused-arguments -fcolor-diagnostics
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang -Qunused-arguments -fcolor-diagnostics accepts -g... yes
checking for clang -Qunused-arguments -fco... (11756 more, please see e.stdout)

  STDERR:

any ideas ?

**Comment by misl6** [comment]
Hi, can you please share a full detailed log?

**Comment by Python3-8** [comment]
Whenever there's any issue about being unable to use `kivy-ios`, I usually tell people to follow these steps: https://tinyurl.com/kivy-ios-setup.

**Comment by jt9897253** [comment]
Same problem here. Looks like it is related to the value returned for `arch`:

`[DEBUG   ] configure: error: Unexpected output of 'arch' on OSX`

On [Apple Silicon](https://en.wikipedia.org/wiki/Apple_silicon), running `arch` on the command line returns `arm64`. On Intel machines, the result value probably was `x86_64` or similar and I guess that's what `hostpython` expects. Thus it looks like this really is a build system error and not related to incorrect usage.

Here's a more verbose log:

```
[DEBUG   ] New State: hostpython3.extract at 2021-07-12 05:18:03.728411
[INFO    ] Build_all hostpython3
[INFO    ] Build hostpython3 for x86_64 (filtered)
[INFO    ] Build hostpython3
[INFO    ] Prebuild hostpython3 for x86_64
[INFO    ] Apply patch pyconfig_detection.patch
[INFO    ] Copy ModulesSetup to Modules/Setup.local
[INFO    ] Build hostpython3 for x86_64
[DEBUG   ] Invoking build_x86_64
[INFO    ] cd /path/to/kivy/build/hostpython3/x86_64/Python-3.8.2/native-build
[INFO    ] Running Shell: /path/to/kivy/build/hostpython3/x86_64/Python-3.8.2/configure ('ac_cv_func_preadv=no', 'ac_cv_func_pwritev=no', 'ac_cv_func_sendfile=no', '--prefix=/path/to/kivy/dist/hostpython3', '--with-openssl=/path/to/kivy/dist/hostopenssl') {'_env': {'TERM_PROGRAM': 'Apple_Terminal', 'SHELL': '/bin/zsh', 'TERM': 'xterm-256color', 'TMPDIR': '/var/folders/3w/4b7gtbg13z7cwxby64bq6w3w0000gn/T/', 'TERM_PROGRAM_VERSION': '440', 'TERM_SESSION_ID': 'B1995198-80B0-42D1-92EA-87ED43BF1E7A', 'USER': 'z', 'SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.wx5hjoJbcX/Listeners', 'PATH': '/path/to/kivy/venv/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/opt/node@14/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/z/.cargo/bin:/Users/z/Projects/go/bin', '__CFBundleIdentifier': 'com.apple.Terminal', 'PWD': '/path/to/kivy', 'XPC_FLAGS': '0x0', 'XPC_SERVICE_NAME': '0', 'SHLVL': '1', 'HOME': '/Users/z', 'LOGNAME': 'z', 'OLDPWD': '/path/to', 'HOMEBREW_PREFIX': '/opt/homebrew', 'HOMEBREW_CELLAR': '/opt/homebrew/Cellar', 'HOMEBREW_REPOSITORY': '/opt/homebrew', 'MANPATH': '/opt/local/share/man:/opt/homebrew/share/man::', 'INFOPATH': '/opt/homebrew/share/info:', 'GOPATH': '/Users/z/Projects/go', 'VMCTLDIR': '/Users/z/Projects/macos/vmcli/vms', 'DISPLAY': ':0', 'VIRTUAL_ENV': '/path/to/kivy/venv', 'PS1': '(venv) %n@%m %1~ %# ', 'LANG': 'de_DE.UTF-8', '_': '/path/to/kivy/venv/bin/toolchain', '__CF_USER_TEXT_ENCODING': '0x1F5:0x0:0x3', 'CC': 'clang -Qunused-arguments -fcolor-diagnostics', 'LDFLAGS': '-lsqlite3 -lffi -L/path/to/kivy/dist/hostlibffi/usr/local/lib -L/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/lib', 'CFLAGS': '--sysroot=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -arch x86_64 -mmacosx-version-min=10.12 -I/path/to/kivy/dist/hostlibffi/usr/local/include -I/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include'}, '_iter': True, '_out_bufsize': 1, '_err_to_out': True}
[DEBUG   ] checking build system type... arm-apple-darwin20.5.0
[DEBUG   ] checking host system type... arm-apple-darwin20.5.0
[DEBUG   ] checking for python3.8... no
[DEBUG   ] checking for python3... python3
[DEBUG   ] checking for --enable-universalsdk... no
[DEBUG   ] checking for --with-universal-archs... no
[DEBUG   ] checking MACHDEP... "darwin"
[DEBUG   ] checking for gcc... clang -Qunused-arguments -fcolor-diagnostics
[DEBUG   ] checking whether the C compiler works... yes
[DEBUG   ] checking for C compiler default output file name... a.out
[DEBUG   ] checking for suffix of executables... 
[DEBUG   ] checking whether we are cross compiling... no
[DEBUG   ] checking for suffix of object files... o
[DEBUG   ] checking whether we are using the GNU C compiler... yes
[DEBUG   ] checking whether clang -Qunused-arguments -fcolor-diagnostics accepts -g... yes
[DEBUG   ] checking for clang -Qunused-arguments -fcolor-diagnostics option to accept ISO C89... none needed
[DEBUG   ] checking how to run the C preprocessor... clang -Qunused-arguments -fcolor-diagnostics -E
[DEBUG   ] checking for grep that handles long lines and -e... /opt/homebrew/bin/ggrep
[DEBUG   ] checking for a sed that does not truncate output... /opt/homebrew/bin/gsed
[DEBUG   ] checking for --with-cxx-main=<compiler>... no
[DEBUG   ] checking for c++... c++
[DEBUG   ] configure:
[DEBUG   ] 
[DEBUG   ]   By default, distutils will build C++ extension modules with "c++".
[DEBUG   ]   If this is not intended, then set CXX on the configure command line.
[DEBUG   ]   
[DEBUG   ] checking for the platform triplet based on compiler characteristics... darwin
[DEBUG   ] checking for -Wl,--no-as-needed... no
[DEBUG   ] checking for egrep... /opt/homebrew/bin/ggrep -E
[DEBUG   ] checking for ANSI C header files... yes
[DEBUG   ] checking for sys/types.h... yes
[DEBUG   ] checking for sys/stat.h... yes
[DEBUG   ] checking for stdlib.h... yes
[DEBUG   ] checking for string.h... yes
[DEBUG   ] checking for memory.h... yes
[DEBUG   ] checking for strings.h... yes
[DEBUG   ] checking for inttypes.h... yes
[DEBUG   ] checking for stdint.h... yes
[DEBUG   ] checking for unistd.h... yes
[DEBUG   ] checking minix/config.h usability... no
[DEBUG   ] checking minix/config.h presence... no
[DEBUG   ] checking for minix/config.h... no
[DEBUG   ] checking whether it is safe to define __EXTENSIONS__... yes
[DEBUG   ] checking for the Android API level... not Android
[DEBUG   ] checking for --with-suffix... 
[DEBUG   ] checking for case-insensitive build directory... yes
[DEBUG   ] checking LIBRARY... libpython$(VERSION)$(ABIFLAGS).a
[DEBUG   ] checking LINKCC... $(PURIFY) $(MAINCC)
[DEBUG   ] checking for GNU ld... no
[DEBUG   ] checking for --enable-shared... no
[DEBUG   ] checking for --enable-profiling... no
[DEBUG   ] checking LDLIBRARY... libpython$(VERSION)$(ABIFLAGS).a
[DEBUG   ] checking for ar... ar
[DEBUG   ] checking for readelf... no
[DEBUG   ] checking for a BSD-compatible install... /opt/homebrew/bin/ginstall -c
[DEBUG   ] checking for a thread-safe mkdir -p... /opt/homebrew/bin/gmkdir -p
[DEBUG   ] checking for --with-pydebug... no
[DEBUG   ] checking for --with-trace-refs... no
[DEBUG   ] checking for --with-assertions... no
[DEBUG   ] checking for --enable-optimizations... no
[DEBUG   ] checking PROFILE_TASK... -m test --pgo
[DEBUG   ] checking for --with-lto... no
[DEBUG   ] checking for llvm-profdata... no
[DEBUG   ] configure: llvm-profdata found via xcrun: /usr/bin/xcrun llvm-profdata
[DEBUG   ] checking for -Wextra... yes
[DEBUG   ] checking whether clang -Qunused-arguments -fcolor-diagnostics accepts and needs -fno-strict-aliasing... no
[DEBUG   ] checking if we can turn off clang -Qunused-arguments -fcolor-diagnostics unused result warning... yes
[DEBUG   ] checking if we can turn off clang -Qunused-arguments -fcolor-diagnostics unused parameter warning... yes
[DEBUG   ] checking if we can turn off clang -Qunused-arguments -fcolor-diagnostics missing field initializers warning... yes
[DEBUG   ] checking if we can turn on clang -Qunused-arguments -fcolor-diagnostics mixed sign comparison warning... yes
[DEBUG   ] checking if we can turn on clang -Qunused-arguments -fcolor-diagnostics unreachable code warning... yes
[DEBUG   ] checking if we can turn on clang -Qunused-arguments -fcolor-diagnostics strict-prototypes warning... yes
[DEBUG   ] checking if we can make implicit function declaration an error in clang -Qunused-arguments -fcolor-diagnostics... yes
[DEBUG   ] checking which MACOSX_DEPLOYMENT_TARGET to use... 11.4
[DEBUG   ] checking whether pthreads are available without options... yes
[DEBUG   ] checking whether c++ also accepts flags for thread support... no
[DEBUG   ] checking for ANSI C header files... (cached) yes
[DEBUG   ] checking asm/types.h usability... no
[DEBUG   ] checking asm/types.h presence... no
[DEBUG   ] checking for asm/types.h... no
[DEBUG   ] checking crypt.h usability... no
[DEBUG   ] checking crypt.h presence... no
[DEBUG   ] checking for crypt.h... no
[DEBUG   ] checking conio.h usability... no
[DEBUG   ] checking conio.h presence... no
[DEBUG   ] checking for conio.h... no
[DEBUG   ] checking direct.h usability... no
[DEBUG   ] checking direct.h presence... no
[DEBUG   ] checking for direct.h... no
[DEBUG   ] checking dlfcn.h usability... yes
[DEBUG   ] checking dlfcn.h presence... yes
[DEBUG   ] checking for dlfcn.h... yes
[DEBUG   ] checking errno.h usability... yes
[DEBUG   ] checking errno.h presence... yes
[DEBUG   ] checking for errno.h... yes
[DEBUG   ] checking fcntl.h usability... yes
[DEBUG   ] checking fcntl.h presence... yes
[DEBUG   ] checking for fcntl.h... yes
[DEBUG   ] checking grp.h usability... yes
[DEBUG   ] checking grp.h presence... yes
[DEBUG   ] checking for grp.h... yes
[DEBUG   ] checking ieeefp.h usability... no
[DEBUG   ] checking ieeefp.h presence... no
[DEBUG   ] checking for ieeefp.h... no
[DEBUG   ] checking io.h usability... no
[DEBUG   ] checking io.h presence... no
[DEBUG   ] checking for io.h... no
[DEBUG   ] checking langinfo.h usability... yes
[DEBUG   ] checking langinfo.h presence... yes
[DEBUG   ] checking for langinfo.h... yes
[DEBUG   ] checking libintl.h usability... no
[DEBUG   ] checking libintl.h presence... yes
[DEBUG   ] configure: WARNING: libintl.h: present but cannot be compiled
[DEBUG   ] configure: WARNING: libintl.h:     check for missing prerequisite headers?
[DEBUG   ] configure: WARNING: libintl.h: see the Autoconf documentation
[DEBUG   ] configure: WARNING: libintl.h:     section "Present But Cannot Be Compiled"
[DEBUG   ] configure: WARNING: libintl.h: proceeding with the compiler's result
[DEBUG   ] configure: WARNING:     ## --------------------------------------- ##
[DEBUG   ] configure: WARNING:     ## Report this to https://bugs.python.org/ ##
[DEBUG   ] configure: WARNING:     ## --------------------------------------- ##
[DEBUG   ] checking for libintl.h... no
[DEBUG   ] checking process.h usability... no
[DEBUG   ] checking process.h presence... no
[DEBUG   ] checking for process.h... no
[DEBUG   ] checking pthread.h usability... yes
[DEBUG   ] checking pthread.h presence... yes
[DEBUG   ] checking for pthread.h... yes
[DEBUG   ] checking sched.h usability... yes
[DEBUG   ] checking sched.h presence... yes
[DEBUG   ] checking for sched.h... yes
[DEBUG   ] checking shadow.h usability... no
[DEBUG   ] checking shadow.h presence... no
[DEBUG   ] checking for shadow.h... no
[DEBUG   ] checking signal.h usability... yes
[DEBUG   ] checking signal.h presence... yes
[DEBUG   ] checking for signal.h... yes
[DEBUG   ] checking stropts.h usability... no
[DEBUG   ] checking stropts.h presence... no
[DEBUG   ] checking for stropts.h... no
[DEBUG   ] checking termios.h usability... yes
[DEBUG   ] checking termios.h presence... yes
[DEBUG   ] checking for termios.h... yes
[DEBUG   ] checking utime.h usability... yes
[DEBUG   ] checking utime.h presence... yes
[DEBUG   ] checking for utime.h... yes
[DEBUG   ] checking poll.h usability... yes
[DEBUG   ] checking poll.h presence... yes
[DEBUG   ] checking for poll.h... yes
[DEBUG   ] checking sys/devpoll.h usability... no
[DEBUG   ] checking sys/devpoll.h presence... no
[DEBUG   ] checking for sys/devpoll.h... no
[DEBUG   ] checking sys/epoll.h usability... no
[DEBUG   ] checking sys/epoll.h presence... no
[DEBUG   ] checking for sys/epoll.h... no
[DEBUG   ] checking sys/poll.h usability... yes
[DEBUG   ] checking sys/poll.h presence... yes
[DEBUG   ] checking for sys/poll.h... yes
[DEBUG   ] checking sys/audioio.h usability... no
[DEBUG   ] checking sys/audioio.h presence... no
[DEBUG   ] checking for sys/audioio.h... no
[DEBUG   ] checking sys/xattr.h usability... yes
[DEBUG   ] checking sys/xattr.h presence... yes
[DEBUG   ] checking for sys/xattr.h... yes
[DEBUG   ] checking sys/bsdtty.h usability... no
[DEBUG   ] checking sys/bsdtty.h presence... no
[DEBUG   ] checking for sys/bsdtty.h... no
[DEBUG   ] checking sys/event.h usability... yes
[DEBUG   ] checking sys/event.h presence... yes
[DEBUG   ] checking for sys/event.h... yes
[DEBUG   ] checking sys/file.h usability... yes
[DEBUG   ] checking sys/file.h presence... yes
[DEBUG   ] checking for sys/file.h... yes
[DEBUG   ] checking sys/ioctl.h usability... yes
[DEBUG   ] checking sys/ioctl.h presence... yes
[DEBUG   ] checking for sys/ioctl.h... yes
[DEBUG   ] checking sys/kern_control.h usability... yes
[DEBUG   ] checking sys/kern_control.h presence... yes
[DEBUG   ] checking for sys/kern_control.h... yes
[DEBUG   ] checking sys/loadavg.h usability... no
[DEBUG   ] checking sys/loadavg.h presence... no
[DEBUG   ] checking for sys/loadavg.h... no
[DEBUG   ] checking sys/lock.h usability... yes
[DEBUG   ] checking sys/lock.h presence... yes
[DEBUG   ] checking for sys/lock.h... yes
[DEBUG   ] checking sys/mkdev.h usability... no
[DEBUG   ] checking sys/mkdev.h presence... no
[DEBUG   ] checking for sys/mkdev.h... no
[DEBUG   ] checking sys/modem.h usability... no
[DEBUG   ] checking sys/modem.h presence... no
[DEBUG   ] checking for sys/modem.h... no
[DEBUG   ] checking sys/param.h usability... yes
[DEBUG   ] checking sys/param.h presence... yes
[DEBUG   ] checking for sys/param.h... yes
[DEBUG   ] checking sys/random.h usability... yes
[DEBUG   ] checking sys/random.h presence... yes
[DEBUG   ] checking for sys/random.h... yes
[DEBUG   ] checking sys/select.h usability... yes
[DEBUG   ] checking sys/select.h presence... yes
[DEBUG   ] checking for sys/select.h... yes
[DEBUG   ] checking sys/sendfile.h usability... no
[DEBUG   ] checking sys/sendfile.h presence... no
[DEBUG   ] checking for sys/sendfile.h... no
[DEBUG   ] checking sys/socket.h usability... yes
[DEBUG   ] checking sys/socket.h presence... yes
[DEBUG   ] checking for sys/socket.h... yes
[DEBUG   ] checking sys/statvfs.h usability... yes
[DEBUG   ] checking sys/statvfs.h presence... yes
[DEBUG   ] checking for sys/statvfs.h... yes
[DEBUG   ] checking for sys/stat.h... (cached) yes
[DEBUG   ] checking sys/syscall.h usability... yes
[DEBUG   ] checking sys/syscall.h presence... yes
[DEBUG   ] checking for sys/syscall.h... yes
[DEBUG   ] checking sys/sys_domain.h usability... yes
[DEBUG   ] checking sys/sys_domain.h presence... yes
[DEBUG   ] checking for sys/sys_domain.h... yes
[DEBUG   ] checking sys/termio.h usability... no
[DEBUG   ] checking sys/termio.h presence... no
[DEBUG   ] checking for sys/termio.h... no
[DEBUG   ] checking sys/time.h usability... yes
[DEBUG   ] checking sys/time.h presence... yes
[DEBUG   ] checking for sys/time.h... yes
[DEBUG   ] checking sys/times.h usability... yes
[DEBUG   ] checking sys/times.h presence... yes
[DEBUG   ] checking for sys/times.h... yes
[DEBUG   ] checking for sys/types.h... (cached) yes
[DEBUG   ] checking sys/uio.h usability... yes
[DEBUG   ] checking sys/uio.h presence... yes
[DEBUG   ] checking for sys/uio.h... yes
[DEBUG   ] checking sys/un.h usability... yes
[DEBUG   ] checking sys/un.h presence... yes
[DEBUG   ] checking for sys/un.h... yes
[DEBUG   ] checking sys/utsname.h usability... yes
[DEBUG   ] checking sys/utsname.h presence... yes
[DEBUG   ] checking for sys/utsname.h... yes
[DEBUG   ] checking sys/wait.h usability... yes
[DEBUG   ] checking sys/wait.h presence... yes
[DEBUG   ] checking for sys/wait.h... yes
[DEBUG   ] checking pty.h usability... no
[DEBUG   ] checking pty.h presence... no
[DEBUG   ] checking for pty.h... no
[DEBUG   ] checking libutil.h usability... no
[DEBUG   ] checking libutil.h presence... no
[DEBUG   ] checking for libutil.h... no
[DEBUG   ] checking sys/resource.h usability... yes
[DEBUG   ] checking sys/resource.h presence... yes
[DEBUG   ] checking for sys/resource.h... yes
[DEBUG   ] checking netpacket/packet.h usability... no
[DEBUG   ] checking netpacket/packet.h presence... no
[DEBUG   ] checking for netpacket/packet.h... no
[DEBUG   ] checking sysexits.h usability... yes
[DEBUG   ] checking sysexits.h presence... yes
[DEBUG   ] checking for sysexits.h... yes
[DEBUG   ] checking bluetooth.h usability... no
[DEBUG   ] checking bluetooth.h presence... no
[DEBUG   ] checking for bluetooth.h... no
[DEBUG   ] checking linux/tipc.h usability... no
[DEBUG   ] checking linux/tipc.h presence... no
[DEBUG   ] checking for linux/tipc.h... no
[DEBUG   ] checking linux/random.h usability... no
[DEBUG   ] checking linux/random.h presence... no
[DEBUG   ] checking for linux/random.h... no
[DEBUG   ] checking spawn.h usability... yes
[DEBUG   ] checking spawn.h presence... yes
[DEBUG   ] checking for spawn.h... yes
[DEBUG   ] checking util.h usability... yes
[DEBUG   ] checking util.h presence... yes
[DEBUG   ] checking for util.h... yes
[DEBUG   ] checking alloca.h usability... yes
[DEBUG   ] checking alloca.h presence... yes
[DEBUG   ] checking for alloca.h... yes
[DEBUG   ] checking endian.h usability... no
[DEBUG   ] checking endian.h presence... no
[DEBUG   ] checking for endian.h... no
[DEBUG   ] checking sys/endian.h usability... no
[DEBUG   ] checking sys/endian.h presence... no
[DEBUG   ] checking for sys/endian.h... no
[DEBUG   ] checking sys/sysmacros.h usability... no
[DEBUG   ] checking sys/sysmacros.h presence... no
[DEBUG   ] checking for sys/sysmacros.h... no
[DEBUG   ] checking linux/memfd.h usability... no
[DEBUG   ] checking linux/memfd.h presence... no
[DEBUG   ] checking for linux/memfd.h... no
[DEBUG   ] checking sys/memfd.h usability... no
[DEBUG   ] checking sys/memfd.h presence... no
[DEBUG   ] checking for sys/memfd.h... no
[DEBUG   ] checking sys/mman.h usability... yes
[DEBUG   ] checking sys/mman.h presence... yes
[DEBUG   ] checking for sys/mman.h... yes
[DEBUG   ] checking for dirent.h that defines DIR... yes
[DEBUG   ] checking for library containing opendir... none required
[DEBUG   ] checking whether sys/types.h defines makedev... yes
[DEBUG   ] checking bluetooth/bluetooth.h usability... no
[DEBUG   ] checking bluetooth/bluetooth.h presence... no
[DEBUG   ] checking for bluetooth/bluetooth.h... no
[DEBUG   ] checking for net/if.h... yes
[DEBUG   ] checking for linux/netlink.h... no
[DEBUG   ] checking for linux/qrtr.h... no
[DEBUG   ] checking for linux/vm_sockets.h... no
[DEBUG   ] checking for linux/can.h... no
[DEBUG   ] checking for linux/can/raw.h... no
[DEBUG   ] checking for linux/can/bcm.h... no
[DEBUG   ] checking for clock_t in time.h... yes
[DEBUG   ] checking for makedev... yes
[DEBUG   ] checking for le64toh... no
[DEBUG   ] checking for mode_t... yes
[DEBUG   ] checking for off_t... yes
[DEBUG   ] checking for pid_t... yes
[DEBUG   ] checking for size_t... yes
[DEBUG   ] checking for uid_t in sys/types.h... yes
[DEBUG   ] checking for ssize_t... yes
[DEBUG   ] checking for __uint128_t... yes
[DEBUG   ] checking size of int... 4
[DEBUG   ] checking size of long... 8
[DEBUG   ] checking size of long long... 8
[DEBUG   ] checking size of void *... 8
[DEBUG   ] checking size of short... 2
[DEBUG   ] checking size of float... 4
[DEBUG   ] checking size of double... 8
[DEBUG   ] checking size of fpos_t... 8
[DEBUG   ] checking size of size_t... 8
[DEBUG   ] checking size of pid_t... 4
[DEBUG   ] checking size of uintptr_t... 8
[DEBUG   ] checking for long double... yes
[DEBUG   ] checking size of long double... 16
[DEBUG   ] checking size of _Bool... 1
[DEBUG   ] checking size of off_t... 8
[DEBUG   ] checking whether to enable large file support... no
[DEBUG   ] checking size of time_t... 8
[DEBUG   ] checking for pthread_t... yes
[DEBUG   ] checking size of pthread_t... 8
[DEBUG   ] checking size of pthread_key_t... 8
[DEBUG   ] checking whether pthread_key_t is compatible with int... no
[DEBUG   ] configure: error: Unexpected output of 'arch' on OSX
Exception in thread background thread for pid 13803:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 1683, in wrap
    fn(*rgs, **kwargs)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 2662, in background_thread
    handle_exit_code(exit_code)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 2349, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 905, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /path/to/kivy/build/hostpython3/x86_64/Python-3.8.2/configure ac_cv_func_preadv=no ac_cv_func_pwritev=no ac_cv_func_sendfile=no --prefix=/path/to/kivy/dist/hostpython3 --with-openssl=/path/to/kivy/dist/hostopenssl

  STDOUT:
checking build system type... arm-apple-darwin20.5.0
checking host system type... arm-apple-darwin20.5.0
checking for python3.8... no
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "darwin"
checking for gcc... clang -Qunused-arguments -fcolor-diagnostics
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang -Qunused-arguments -fcolor-diagnostics accepts -g... yes
checking for clang -Qunused-arguments -fco... (12302 more, please see e.stdout)

  STDERR:

[INFO    ] cd /path/to/kivy/build/hostpython3/x86_64/Python-3.8.2
Traceback (most recent call last):
  File "/path/to/kivy/venv/bin/toolchain", line 8, in <module>
    sys.exit(main())
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 1519, in main
    ToolchainCL()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 1276, in __init__
    getattr(self, args.command)()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 1337, in build
    build_recipes(args.recipe, ctx)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 1127, in build_recipes
    recipe.execute()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 699, in execute
    self.build_all()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 73, in _cache_execution
    f(self, *args, **kwargs)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 793, in build_all
    self.build(arch)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 73, in _cache_execution
    f(self, *args, **kwargs)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 780, in build
    self.build_arch(arch)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 832, in build_arch
    getattr(self, build)()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/recipes/hostpython3/__init__.py", line 75, in build_x86_64
    shprint(configure,
  File "/path/to/kivy/venv/lib/python3.9/site-packages/kivy_ios/toolchain.py", line 55, in shprint
    for line in cmd:
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 953, in next
    self.wait()
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 879, in wait
    self.handle_command_exit_code(exit_code)
  File "/path/to/kivy/venv/lib/python3.9/site-packages/sh.py", line 905, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /path/to/kivy/build/hostpython3/x86_64/Python-3.8.2/configure ac_cv_func_preadv=no ac_cv_func_pwritev=no ac_cv_func_sendfile=no --prefix=/path/to/kivy/dist/hostpython3 --with-openssl=/path/to/kivy/dist/hostopenssl

  STDOUT:
checking build system type... arm-apple-darwin20.5.0
checking host system type... arm-apple-darwin20.5.0
checking for python3.8... no
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "darwin"
checking for gcc... clang -Qunused-arguments -fcolor-diagnostics
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang -Qunused-arguments -fcolor-diagnostics accepts -g... yes
checking for clang -Qunused-arguments -fco... (12302 more, please see e.stdout)

  STDERR:
```

**Comment by sqwatato** [comment]
I fixed this on my M1 mac by using Rosetta 2. Search up how to download Rosetta 2. Then duplicate terminal and rename one of them as Rosetta(or what ever you want to call it). Then go into finder and find the terminal that you renamed and right click and then choose get info. There you can select open with Rosetta.  There are a lot of issues with switching from intel based chips to the m1 and Rosetta is meant to convert that stuff until more universal thing are made. Email sqwatato@gmail.com if you need more help. (btw im not an expert on this i just had the same problem before)

**Comment by misl6** [comment]
Hi all,
The issue has been fixed via #660.
